### PR TITLE
HELM-293: support string properties in performance datasource

### PR DESCRIPTION
**Pull request by swachter**
Issue: https://issues.opennms.org/browse/HELM-293

This PR adds support for querying string properties to the performance datasource.

@Bonrob2 I also added some documentation; please review.

Open question: I added some help text into `query.options.html` but did not find where this would show up when using the performance datasource. Is the help functiionality broken?

### CODE REVIEW

File: docs/modules/datasources/pages/performance_datasource.adoc [hunk 1]
```diff
@@ -31,6 +31,20 @@ The query is used to identify which attribute (or series) is being queried.
                        Must be unique for all attributes and expression results in the query.
 |===
 
+=== String property queries
+
+Used to read the last value of string valued properties that OpenNMS collects and stores. The query is used to identify which string property is queried. A panel can contain multiple string property queries. The result can be visualized in a table. Grafana's transforms like "Filter by name", "Organize Fields", or "Merge" can be used to adjust the visualized table as needed.
```
Comment (Bonrob2) [inline docs/modules/datasources/pages/performance_datasource.adoc:36]:
```suggestion
Used to read the last value of string-valued properties that OpenNMS collects and stores.
The query identifies which string property is queried. 
A panel can contain multiple string property queries. 
The result can be visualized in a table. 
Use Grafana's transforms like "Filter by name", "Organize Fields", or "Merge" to adjust the visualized table as needed.
```
Each sentence needs to be on a separate line.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Looks good, Stefan. Minor changes.

**Review by RangerRick — APPROVED** [review]
this looks pretty good to me 👍

File: docs/modules/datasources/pages/performance_datasource.adoc [hunk 2]
```diff
@@ -31,6 +31,20 @@ The query is used to identify which attribute (or series) is being queried.
...
+
+[options="header, autowidth"]
+|===
+| Name               | Description
+| Node               | Node ID of the node to query.
+Can also be a variable with a `nodeFilter()` query.
+| Resource           | Resource ID containing the requested attribute.
+Can also be a variable with a `nodeResources()` query.
+| String Property    | Name of the attribute to query.
+|===
```
Comment (mmahacek) [inline docs/modules/datasources/pages/performance_datasource.adoc:46]:
```suggestion
|===
| Name
| Description

| Node
| Node ID of the node to query.
Can also be a variable with a `nodeFilter()` query.

| Resource
| Resource ID containing the requested attribute.
Can also be a variable with a `nodeResources()` query.

| String Property
| Name of the attribute to query.
|===
```

**Review by RangerRick — APPROVED** [review]

**Review by Bonrob2 — DISMISSED** [review]
LGTM!

**Review by mmahacek — CHANGES_REQUESTED** [review]
Modified Signature error with deb on Ubuntu with Grafana 8.3.3

**Comment by mmahacek** [comment]
And the ZIP from the latest build running on 8.3.3 in Docker has an issue where UI elements aren't displayed. 
image

**Review by RangerRick — CHANGES_REQUESTED** [review]
please rebase this to `release-7.x`

**Comment by swachter** [comment]
rebased on 7.x; please review again

**Review by mmahacek — APPROVED** [review]
This works for me on Grafana 8.3.3 on both Ubuntu and Docker.

**Review by RangerRick — DISMISSED** [review]

**Comment by swachter** [comment]
I am sorry, I have to make another small change. Simply throwing an error in case that string property queries are mixed with other kinds of queries does not yield to an error message in the view. The fix is to return a proper response with an error message.

**Review by RangerRick — APPROVED** [review]
ah yeah, good catch

**Pull request merged**